### PR TITLE
ARO-12469: Added etchost files from the ARO fork 

### PR DESCRIPTION
**Pull request by rna-afk**
Adding the code to generate etchost files locally for ignition.

### CODE REVIEW

File: pkg/installer/etchost/etchost_assets.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	etcHostMasterMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "master")
+	if err != nil {
+		return err
+	}
+	etcHostWorkerMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "worker")
+	if err != nil {
+		return err
+	}
+
+	err = dnsmasq.AppendMachineConfigToBootstrap(etcHostMasterMachineConfig, bootstrapAsset, "/opt/openshift/openshift/99_openshift-machineconfig_99-master-aro-etc-hosts-gateway-domains.yaml")
```
Comment (rna-afk) [inline pkg/installer/etchost/etchost_assets.go:25]: Reuses functions from the dnsmasq PR 
https://github.com/openshift/installer-aro-wrapper/blob/d6af50321aced4a322a645539afa39420b679469/pkg/installer/dnsmasq/dnsmasq_config.go#L40

File: pkg/installer/etchost/etchost_config.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	etcHostMasterMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "master")
+	if err != nil {
+		return err
+	}
+	etcHostWorkerMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "worker")
+	if err != nil {
+		return err
+	}
+
+	err = dnsmasq.AppendMachineConfigToBootstrap(etcHostMasterMachineConfig, bootstrapAsset, "/opt/openshift/openshift/99_openshift-machineconfig_99-master-aro-etc-hosts-gateway-domains.yaml")
```
Comment (rna-afk) [inline pkg/installer/etchost/etchost_config.go:25]: Reuses functions from the dnsmasq PR 
https://github.com/openshift/installer-aro-wrapper/blob/d6af50321aced4a322a645539afa39420b679469/pkg/installer/dnsmasq/dnsmasq_config.go#L40

Comment (rna-afk) [reply pkg/installer/etchost/etchost_config.go]: Moved them to the util file.

**Comment by zaneb** [comment]
Need to resolve the merge conflict with your other PR that just merged.

**Comment by rna-afk** [comment]
Rebased the PR. Should be good to go.

**Pull request merged**